src/viking.js

```diff
@@ -1,19 +1,113 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack(){
+    return this.strength;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage;    
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+  constructor(name, health, strength) {
+  super(health, strength)
+  this.name = name;
+  }
+  
+  receiveDamage(damage) {
+    super.receiveDamage(damage);
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    }
+    else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
```

Comment:
nesses casos vc nem precisa usar "else" se quiser, depois do if se quiser mandar direto o "return ${this.name} has died in act of combat", também da certo 😉

Reply:
Vdd! Vlw o toque